src/core/diff.ts

```diff
@@ -26,7 +26,7 @@ export function reference(previousProperty: any, newProperty: any): PropertyChan
 	};
 }
 
-export function shallow(previousProperty: any, newProperty: any): PropertyChangeRecord {
+export function shallow(previousProperty: any, newProperty: any, depth = 0): PropertyChangeRecord {
```

Comment:
Adds support to choose the depth of the diff... defaults to 0 which is the equivalent of the current behaviour